tracing-opentelemetry/src/main/java/org/axonframework/tracing/opentelemetry/OpenTelemetrySpanFactory.java

```diff
@@ -0,0 +1,186 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.tracing.opentelemetry;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.messaging.Message;
+import org.axonframework.tracing.Span;
+import org.axonframework.tracing.SpanAttributesProvider;
+import org.axonframework.tracing.SpanFactory;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.axonframework.tracing.SpanUtils.determineMessageName;
+
+/**
+ * Creates {@link Span} implementations that are compatible with OpenTelemetry java agent instrumentation. OpenTelemetry
+ * is a standard to collect logging, tracing and metrics from applications. This {@link SpanFactory} focusses on
+ * supporting the tracing part of the standard.
+ * <p>
+ * To get started with OpenTelemetry, <a href="https://opentelemetry.io/docs/">check out their documentation</a>.
+ */
+public class OpenTelemetrySpanFactory implements SpanFactory {
+
+    private final Tracer tracer = GlobalOpenTelemetry.getTracer("axon-opentelemetry");
+    private final List<SpanAttributesProvider> spanAttributesProviders;
+    private final boolean useParentsInsteadOfLinks;
+    private final TextMapPropagator textMapPropagator = GlobalOpenTelemetry.getPropagators().getTextMapPropagator();
+
+    /**
+     * Creates a new {@link SpanFactory} that creates {@link Span} implementations that are compatible with
+     * OpenTelemetry.
+     *
+     * @param spanAttributesProviders A list of {@link SpanAttributesProvider}s, which add metadata to span based on the
+     *                                input of a {@link Message}.
+     */
+    public OpenTelemetrySpanFactory(List<SpanAttributesProvider> spanAttributesProviders) {
+        this(spanAttributesProviders, false);
+    }
+
+    /**
+     * Creates a new {@link SpanFactory} that creates {@link Span} implementations that are compatible with
+     * OpenTelemetry.
+     *
+     * @param spanAttributesProviders  A list of {@link SpanAttributesProvider}s, which add metadata to span based on
+     *                                 the input of a {@link Message}.
+     * @param useParentsInsteadOfLinks To make traces comprehensible, we can split them up and link them together
+     *                                 instead of forming one big trace. Supplying true as an argument will create large
+     *                                 traces instead.
+     */
+    public OpenTelemetrySpanFactory(List<SpanAttributesProvider> spanAttributesProviders,
```

Comment:
Although we're on two parameter right now, this might change in the future. 
To not get an explosion of constructors, I think it would be more helpful to use the builder-pattern for this class too.
That gives us flexibility with adding and deprecating parameters without increasing the number of constructors.